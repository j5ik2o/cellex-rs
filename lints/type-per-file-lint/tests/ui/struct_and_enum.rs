#![feature(register_tool)]
#![register_tool(type_per_file)]
#![warn(type_per_file::multiple_type_definitions)]

pub struct SampleStruct;

pub enum SampleEnum {
  Unit,
}

fn main() {}
