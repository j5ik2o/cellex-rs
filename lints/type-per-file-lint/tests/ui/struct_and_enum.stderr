warning: 構造体・列挙型・トレイトはファイルごとに1つまでにしてください
  --> $DIR/struct_and_enum.rs:7:1
   |
LL | / pub enum SampleEnum {
LL | |   Unit,
LL | | }
   | |_^
   |
note: このファイルでは既に `SampleStruct` (struct) が定義されています
  --> $DIR/struct_and_enum.rs:5:1
   |
LL | pub struct SampleStruct;
   | ^^^^^^^^^^^^^^^^^^^^^^^^
   = note: 対象ファイル: $DIR/struct_and_enum.rs
   = note: 検出された型一覧: SampleStruct (struct), SampleEnum (enum)
   = help: 型ごとに別ファイルを作成し、親モジュールで `mod` 宣言を更新してください
   = note: AI向けアドバイス: 1. `struct_and_enum.rs` と同じディレクトリに `sample_enum.rs` を新規作成しこの `SampleEnum` を移動（サブディレクトリを掘らず `.rs` ファイル単体で配置） 2. 親モジュールで `mod sample_enum;` を追加 3. 必要なら `pub use` を調整して `cargo check` を実行してください。
   = note: `#[warn(multiple_type_definitions)]` on by default

warning: 1 warning emitted

